lib/Parser/Parse.cpp

```diff
@@ -4985,6 +4985,12 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
 
         this->ParseFncFormals<buildAST>(pnodeFnc, flags);
 
+        if ((pnodeFnc->sxFnc.CallsEval() || pnodeFnc->sxFnc.ChildCallsEval())
+                && !m_scriptContext->GetConfig()->IsES6DefaultArgsSplitScopeEnabled())
+        {
+                Error(ERREvalNotSupportedInParamScope);
+        }
```

Comment:
Could this go in ParserFncFormals instead?  Seems weird to be here in this higher level code.